### PR TITLE
add .env.database for database and docker for development

### PR DESCRIPTION
**Pull request by hpbonfim**
I've read the #28 and #40 issues and code about it.
Here's the solution:
- Add a .env.database for database environments
- Add a Dockerfile for create a image of the Dolos
- Create a docker-compose.yml to development.
- Add Portainer to manage the containers.
- Add MySQL and phpMyAdmin to manage the local database.
* Add the Dolos inside the container for fast development with mysql.
- Update the README.md and write how to use all those tools.

### CODE REVIEW

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]
Nice additions, but do we have an issue open to add docker support? Please create one.

**Comment by hpbonfim** [comment]
Ok, I created one for this support #53

File: .env.database [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+DB_ROOT_PASSWORD=
+DB_HOST=
+
+# For docker database (Don't change the HOST=mysql because this is the docker service)
+DOCKER_DB_DATABASE=
+DOCKER_DB_PORT=
+DOCKER_DB_USER=
+DOCKER_DB_PASSWORD=
+DOCKER_DB_ROOT_PASSWORD=
+DOCKER_DB_HOST=mysql
```
Comment (YashKumarVerma) [inline .env.database:15]: should we move this to .env.example ?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN npm install --quiet
+
+RUN npm install nodemon -g
+
+COPY . .
+
+EXPOSE 3000
+
+CMD nodemon -L --watch . src/app.js
```
Comment (YashKumarVerma) [inline Dockerfile:17]: why running a development command in docker container ? we won't be hot reloading right ?

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]

File: .env.database (hunk 1 above)
Comment (hpbonfim) [reply .env.database:15]: Yes, I don't move because other people edit this file

File: Dockerfile (hunk 2 above)
Comment (hpbonfim) [reply Dockerfile:17]: When we create a dockerfile, the docker build the same app but with completely different location to keep all services together and we need to be able to sync this location with src local files. 
There is not difference between this command or npm run dev, but I keep more simple for other people learning this.

**Review by YashKumarVerma — APPROVED** [review]

**Comment by YashKumarVerma** [comment]
please resolve merge conflicts.

**Review by hpbonfim — COMMENTED** [review]
⠀

**Comment by lopeselio** [comment]
#50 merged

**Pull request merged**